### ISSUE TITLE
Improve Task instrumentation

### ISSUE THREAD

**Issue by tombruijn** [appsignal/appsignal-elixir#183]
Our Task instrumentation is currently lacking. The reports that come in provide a minimal stacktrace and a slightly too generic error message.

Still need to reproduce this. 

When a Task crashes on it own, not related to #173, but similar type error, we only get minimal information from the Elixir error reporter. Can we improve this?

![pasted image at 2017_03_17 09_40](https://cloud.githubusercontent.com/assets/282402/24036180/bea68d76-0af8-11e7-89b8-af59e1a65623.png)

Reported in: https://app.intercom.io/a/apps/yzor8gyw/inbox/conversation/8828158260

**Comment by jeffkreeftmeijer** [comment]
After having merged #187 and #193, this error will now look like this on Appsignal.com:

![screen shot 2017-04-06 at 08 48 45](https://cloud.githubusercontent.com/assets/43621/24741469/59500b4c-1aa6-11e7-9618-33024ba9cb59.png)

To reproduce:

``` elixir
:proc_lib.spawn(fn() ->
  Task.async(fn() ->
    Process.sleep(2)
  end)
  |> Task.await(1)
end)
```

**Comment by tombruijn** [comment]
Fixed by #357
